string not assignable to interp.valueInterface

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

func Sprintf(format string, a ...interface{}) string {
    return fmt.Sprintf(format, a...)
}

func main() {
   fmt.Println(Sprintf("Hello %s", "World!"))
}
```

Expected result:
```console
$ go run ./sample.go
Hello World!
```

Got:
```console
$ yaegi ./sample.go
./sample.go:10:4: panic
reflect.Set: value of type string is not assignable to type interp.valueInterface
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc00033c600, 0x40)
        /share/opt/go/default/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc0000b9df0)
        /users/myuser/go/pkg/mod/github.com/containous/yaegi@v0.8.8/interp/interp.go:330 +0xbc
panic(0xca5420, 0xc0000707e0)
        /share/opt/go/default/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc00037e0a0, 0xc000388500)
        /users/myuser/go/pkg/mod/github.com/containous/yaegi@v0.8.8/interp/run.go:115 +0x20c
panic(0xca5420, 0xc0000707e0)
        /share/opt/go/default/src/runtime/panic.go:969 +0x166
reflect.Value.assignTo(0xca5420, 0xc0000706b0, 0x98, 0xdd8058, 0xb, 0xd28c40, 0x0, 0x97, 0xc00000ef80, 0xc99000)
        /share/opt/go/default/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xd28c40, 0xc00000efa0, 0x199, 0xca5420, 0xc0000706b0, 0x98)
        /share/opt/go/default/src/reflect/value.go:1532 +0xbd
reflect.Append(0xc99000, 0xc00000ef20, 0x197, 0xc0000b9bd8, 0x1, 0x1, 0xc0000b9aa0, 0x411918, 0xc000197797)
        /share/opt/go/default/src/reflect/value.go:2028 +0xe8
github.com/containous/yaegi/interp.call.func7(0xc00037e0a0, 0xc00037b730)
        /users/myuser/go/pkg/mod/github.com/containous/yaegi@v0.8.8/interp/run.go:922 +0xaca
github.com/containous/yaegi/interp.runCfg(0xc000388500, 0xc00037e0a0)
        /users/myuser/go/pkg/mod/github.com/containous/yaegi@v0.8.8/interp/run.go:121 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0002941a0, 0xc000385d00, 0xc00037e000)
        /users/myuser/go/pkg/mod/github.com/containous/yaegi@v0.8.8/interp/run.go:99 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0002941a0, 0xc0003321a0, 0xc5, 0x0, 0x0, 0x0, 0x0, 0x0)
        /users/myuser/go/pkg/mod/github.com/containous/yaegi@v0.8.8/interp/interp.go:406 +0x52b
main.main()
        /users/myuser/go/pkg/mod/github.com/containous/yaegi@v0.8.8/cmd/yaegi/yaegi.go:143 +0x553
```

Version: v0.8.8

NOTE:  This issue seems similar to the following Fixed issues:
* [https://github.com/containous/yaegi/issues/589](https://github.com/containous/yaegi/issues/589)
* [https://github.com/containous/yaegi/issues/527](https://github.com/containous/yaegi/issues/527)

--- a/_test/variadic9.go
+++ b/_test/variadic9.go
@@ -1,0 +1,14 @@
+package main
+
+import "fmt"
+
+func Sprintf(format string, a ...interface{}) string {
+	return fmt.Sprintf(format, a...)
+}
+
+func main() {
+	fmt.Println(Sprintf("Hello %s", "World!"))
+}
+
+// Output:
+// Hello World!

--- a/interp/run.go
+++ b/interp/run.go
@@ -801,17 +801,18 @@ func call(n *node) {
 				values = append(values, func(f *frame) reflect.Value { return f.data[ind] })
 			}
 		default:
+			var arg *itype
+			if variadic >= 0 && i >= variadic {
+				arg = n.child[0].typ.arg[variadic].val
+			} else {
+				arg = n.child[0].typ.arg[i]
+			}
 			if c.kind == basicLit || c.rval.IsValid() {
-				var argType reflect.Type
-				if variadic >= 0 && i >= variadic {
-					argType = n.child[0].typ.arg[variadic].val.TypeOf()
-				} else {
-					argType = n.child[0].typ.arg[i].TypeOf()
-				}
+				argType := arg.TypeOf()
 				convertLiteralValue(c, argType)
 			}
 			switch {
-			case len(n.child[0].typ.arg) > i && n.child[0].typ.arg[i].cat == interfaceT:
+			case arg.cat == interfaceT:
 				values = append(values, genValueInterface(c))
 			case isRecursiveType(c.typ, c.typ.rtype):
 				values = append(values, genValueRecursiveInterfacePtrValue(c))
@@ -1072,7 +1073,7 @@ func callBin(n *node) {
 				values = append(values, genFunctionWrapper(c))
 			case interfaceT:
 				values = append(values, genValueInterfaceValue(c))
-			case arrayT:
+			case arrayT, variadicT:
 				switch c.typ.val.cat {
 				case interfaceT:
 					values = append(values, genValueInterfaceArray(c))

